src/NUnitFramework/framework/Constraints/ExactCountConstraint.cs

```diff
@@ -31,9 +31,19 @@ namespace NUnit.Framework.Constraints
     /// item in a collection, succeeding only if a specified
     /// number of items succeed.
     /// </summary>
-    public class ExactCountConstraint : PrefixConstraint
+    public class ExactCountConstraint : Constraint
```

Comment:
That is interesting. I wonder if we could manage it so that the count or existence check would be in one constraint class and the additional constraint modification would be in another class, and they could be composed so that the class that applies an additional constraint does not duplicate any of the count or exists logic.